CDS Maps/Individual Maps/FHIR-to-CDS/nNGM-FHIR-to-CDS-Operation.map

```diff
@@ -1,282 +1,306 @@
+group TransformOperationTherapieCTS(source entry: BackboneElement, target tgt: BackboneElement)
 {
-    src -> tgt.crfid = 'OP1';
-    src -> tgt.type = 'save';
-
-    src.entry as entry then
+    /* ------------------------------ Procedure ---------------------------- */
+    entry.resource as procedure where "resource is Procedure and resource.meta.profile = 'http://uk-koeln.de/fhir/StructureDefinition/Procedure/nNGM/OP'" then
     {
-        /* ------------------------------ Procedure ---------------------------- */
-        entry.resource as procedure where "resource is Procedure and resource.meta.profile = 'http://uk-koeln.de/fhir/StructureDefinition/Procedure/nNGM/OP'" then
+        // Operationsdatum
+        procedure -> tgt.data as data then
         {
-            // Operationsdatum
-            procedure -> tgt.data as data then
+            procedure.performedDateTime as performedDateTime then
             {
-                procedure.performedDateTime as performedDateTime then
-                {
-                    procedure -> data.blockindex = 0;
-                    procedure -> data.groupindex = 0;
-                    procedure -> data.itemid = 'id_842';
-                    performedDateTime -> data.values as values, values.value = performedDateTime;
-                };
+                procedure -> data.blockindex = 0;
+                procedure -> data.groupindex = 0;
+                procedure -> data.itemid = 'id_842';
+                performedDateTime -> data.values as values, values.value = performedDateTime;
             };
-            
-            // Durchführende Einrichtung
-            procedure -> tgt.data as data then
+        };
+        
+        // Durchführende Einrichtung
+        procedure -> tgt.data as data then
+        {
+            procedure.performer as performer, performer.actor as actor, actor.display as display then
             {
-                procedure.performer as performer, performer.actor as actor, actor.display as display then
-                {
-                    procedure -> data.blockindex = 0;
-                    procedure -> data.groupindex = 0;
-                    procedure -> data.itemid = 'id_1162';
-                    display -> data.values as values, values.value = display;
-                };  
-            };
-            
-            // Operative Intention
-            procedure -> tgt.data as data then
+                procedure -> data.blockindex = 0;
+                procedure -> data.groupindex = 0;
+                procedure -> data.itemid = 'id_1162';
+                display -> data.values as values, values.value = display;
+            };  
+        };
+        
+        // Operative Intention
+        procedure -> tgt.data as data then
+        {
+            procedure.reasonCode as reasonCode, reasonCode.coding as coding where "coding.system = 'http://uk-koeln.de/fhir/CodeSystem/nngm/intention-therapie'" then
```

Comment:
The blocks in this function could be simplified. The value is always the code except for 'N/A'. So basically, if it is != N/A, you could just map the value.

Reply:
fixed!